adr/0024-starlight.md

```diff
@@ -0,0 +1,76 @@
+# 24. Migrate documentation to Starlight
+
+Date: 2024-03-30
+
+## Status
+
+Accepted
+
+## Context
+
+The current documentation site is built with Docusaurus 2. While Docusaurus has served us well, it has some limitations, primarily around performance and dependency management.
+
+### Proposed Solutions
+
+1. [**Starlight**](https://starlight.astro.build/) is a new documentation site generator that is built with performance in mind. It is a lightweight, fast, and flexible static site generator powered by [Astro](https://astro.build) that is designed to be easy to use and extend.
+
+    - **Pros**:
+        - Fast performance.
+        - Easy to extend with custom components in any framework.
+        - Built-in support for MDX.
+        - Simplified build process.
+        - Better control over dependencies.
+        - Client-side static site search built-in.
+        - Excellent default theme.
+    - **Cons**:
+        - Newer project with a smaller ecosystem compared to others.
+        - Much fewer plugins and themes available compared to other static site generators.
+        - Only absolute URLs are supported for cross-site links (relative image URLs are supported).
+
+2. [**Hugo**](https://gohugo.io/): is another popular static site generator that is known for its speed and flexibility. It is written in Go and has a large ecosystem of themes and plugins.
```

Comment:
There's a pending PR to convert the site to Hugo that only requires merging and running the conversion script. This would bring Zarf in alignment with all of the other product doc sites.

Reply:
Because Zarf hasn't been converted, there's no preview available unless you pull the branch down and run the conversion locally. However, here's the [Defense Unicorns theme documentation](https://starlit-valkyrie-7f1dd9.netlify.app/docs/) along with it being used for [Pepr](https://app.netlify.com/sites/pepr-docs/), [UDS](https://main--uds.netlify.app/docs/), [Leapfrog](https://docs.leapfrog.ai/docs/), and [Software Factory](https://software-factory-docs.netlify.app/docs/)